### PR TITLE
Fix Courses (endpoints & queries)

### PR DESCRIPTION
**Pull request by marianysilva**
## Summary by CodeRabbit

- **New Features**
	- Added a `DeletedCourse` struct to represent deleted courses.
	- Modified response structures in endpoints to return course information in a unified format.
- **Refactor**
	- Standardized parameter names across functions for better clarity and consistency.
	- Updated function signatures to accept `Course` type for improved data handling.
- **Documentation**
	- Updated and simplified course-related response structures for better user understanding.

### CODE REVIEW

File: internal/course/domain/course_repository.go [hunk 1]
```diff
@@ -1,11 +1,14 @@
 package domain
 
-import "github.com/google/uuid"
+import (
+	"github.com/google/uuid"
+	"github.com/sumelms/microservice-course/internal/shared"
```
Comment (rluders) [inline internal/course/domain/course_repository.go:5]: Here we are creating an external dependency for the domain, but I don't think that this is a problem. There is nothing on DDD that says that we can't do that. Typically, however, you will always desire to have your domain as independent as possible. But, it is just a comment... in this specific case, I don't think that it is a problem.

File: internal/course/endpoints/find_course.go [hunk 2]
```diff
@@ -46,20 +36,13 @@ func makeFindCourseEndpoint(s domain.ServiceInterface) endpoint.Endpoint {
...
 		return &findCourseResponse{
-			UUID:        c.UUID,
-			Code:        c.Code,
-			Name:        c.Name,
-			Underline:   c.Underline,
-			Image:       c.Image,
-			ImageCover:  c.ImageCover,
-			Excerpt:     c.Excerpt,
-			Description: c.Description,
+			Course: &course,
```
Comment (rluders) [inline internal/course/endpoints/find_course.go:45]: Oh, now I understand what you are talking about. So, actually, there are a few things that we need to consider when we use this **direct object return** approach. ChatGPT helped me to organize the comparison:

#### Comparison:

##### Original Code (Object Mapping):
- **Pros**:
  1. **Control over Response Structure**: Allows explicit control over which fields of the domain object are included in the response, enabling customization based on API requirements.
  2. **Flexibility**: Offers flexibility to adjust the structure of the HTTP response independently from the domain object, facilitating maintenance and evolution.
  3. **Encapsulation**: Helps maintain encapsulation by hiding internal details of the domain object from the API contract.

- **Cons**:
  1. **Redundancy**: Requires repetitive mapping of fields from the domain object to the response object, leading to code duplication and potential for errors.
  2. **Complexity**: Increases code complexity and maintenance overhead, especially as the number of endpoints and domain objects grows.
  3. **Risk of Inconsistency**: Possibility of inconsistencies between the domain model and the API contract if mappings are not kept up to date.

##### Proposed Change (Direct Object Return):
- **Pros**:
  1. **Reduced Redundancy**: Eliminates the need for explicit mapping, resulting in cleaner and more concise code.
  2. **Simplicity**: Simplifies codebase by directly returning the domain object, reducing complexity and potential for errors.
  3. **Less Prone to Errors**: Directly extracting data from the domain object minimizes the chances of inconsistencies between the domain model and the API contract.

- **Cons**:
  1. **Limited Control**: Provides less control over the structure of the HTTP response. All fields from the domain object will be exposed, potentially exposing internal details.
  2. **Potential Leakage of Internal Details**: Exposing the domain object directly may reveal internal implementation details, violating encapsulation principles.
  3. **Difficulty in Evolving Response Structure**: Changes to the response structure may require modifications to the domain object, leading to tighter coupling between the API contract and the domain model.

Comment (rluders) [reply internal/course/endpoints/find_course.go:45]: So, the point here is that both approaches have some pros and cons, as usual... I don't have the right answer here, but I'm more inclined to use **object mapping**, 'cause I think that the pros are more beneficial and the cons for that approach can be easily mitigated by tests, in the other hand, the cons from **direct object return** could limit us and also leave to some undesirable data leak.

@arianerocha WDYT?

Comment (marianysilva) [reply internal/course/endpoints/find_course.go:45]: II and III cons of `Proposed Change (Direct Object Return)` changed my mind: **I'm more inclined to use object mapping**

File: internal/course/domain/course_repository.go (hunk 1 above)
Comment (marianysilva) [reply internal/course/domain/course_repository.go]: https://github.com/sumelms/microservice-course/pull/26/commits/55dd95b48f589a3da02aee513c163518f3e84f1a

File: internal/course/endpoints/find_course.go (hunk 2 above)
Comment (marianysilva) [reply internal/course/endpoints/find_course.go]: https://github.com/sumelms/microservice-course/pull/26/commits/55dd95b48f589a3da02aee513c163518f3e84f1a

**Review by rluders — APPROVED** [review]
lgtm

**Pull request merged**